### ISSUE TITLE
Off-by-one error in Range request (byte serving) example

### ISSUE THREAD

**Issue by infrahead** [oven-sh/bun#3087]
### What is the type of issue?

Example code is not working

### What is the issue?

The JS slice method is "between" the indexes, but bytes returned needs to include that last byte!

```
// HTTP range requests are inclusive, but the end param of slice is not, i.e. +1
return new Response(bigFile.slice(start, end+1));
```

### Where did you find it?

https://bun.sh/docs/api/http#streaming-files

**Comment by Jarred-Sumner** [comment]
Are you seeing a bug related to this in code you ran?

This is confusing without more context on Bun, but `Bun.file(path).slice()` is not an Array or TypedArray slice.

it is [`Blob.prototype.slice`](https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice#parameters), which is not between the indices.

<img width="323" alt="image" src="https://github.com/oven-sh/bun/assets/709451/30ca48ad-8837-4f00-b04f-366bbfa252c3">

**Comment by infrahead** [comment]
That is confusing, because yes it was a bug in my code until I changed the index+1, which I thought was simple oversight!  It took an embarrassingly long time troubleshooting a compression library giving an unexpected EOF error, before I realized what was happening. 

Here is a simple example I showing the issue, using the exact snippet from the streaming-files docs (localhost:3000), and then using `npx http-server` to serve same static file (localhost:8080).  10951 is the correct bytes

<img width="736" alt="image" src="https://github.com/oven-sh/bun/assets/7755394/fc6c2068-8680-4bbd-bd65-5c36e7d4bde9">

**Comment by infrahead** [comment]
@Jarred-Sumner Learning about blob, but I just read the page you linked, and in fact it does say about `end`...
 _"An index into the Blob indicating the first byte that will *not* be included in the new Blob (i.e. the byte exactly at this index is not included)."_

I think in your first example, it needed to be `['a' 'b', 'c'].slice(0, 2).length` which is also 2, i.e  both of these return indices 0 and 1.

**Comment by Electroid** [comment]
This was fixed #6252
